### DISCUSSION TITLE
Is the Shopify integration of BTC Pay Server still working?

### DISCUSSION THREAD

**Discussion by RunyCalmera** [btcpayserver/btcpayserver#6494] [Technical Support]
Hi I'm new to BTC Pay Server and Shopify.

I followed all the instructions in the video https://docs.btcpayserver.org/Shopify/

1. My connection of BTC Pay server Aruba (btc.aw) to my shopify store is working. Admin key and API has been set correctly.
2. But when I try to test a payment to my store, I get the following error: Could not initiate BTCPay Server payment method, try again later.

When I try later this is still showing up.

My store I'm using is at: https://1zkv3h-um.myshopify.com

Thanks,

Runy

**Comment by pavlenex** [comment]
@ndeet do you know what's goind on here, seems like some sort of domain error?
<img width="1440" alt="Screenshot 2024-12-16 at 15 10 40" src="https://github.com/user-attachments/assets/23c162b1-995c-418d-8d57-40a812621f69" />

**Reply by RunyCalmera** [reply]
**Some background on that domain:**
When you are in the first 3 days of testing shopify (and you paid for a subscription) and you don't yet have a domain connected to shopify, shopify gives you these kind of domains: https://1zkv3h-um.myshopify.com/.

When you click on that link you get a working shopify website.

<img width="1773" alt="Shopify shop" src="https://github.com/user-attachments/assets/2d15aafb-1e5b-4db6-9679-b126906ad4d4" />

**Reply by pavlenex** [reply]
@RunyCalmera In your btcpay, can you manually generate an invoice? What is the currency that you use? NAG? i don't think we have support for that one tbh. Which rate source do you use?

**Reply by RunyCalmera** [reply]
![image](https://github.com/user-attachments/assets/d0409054-74df-4852-8d1d-e86f83032162)
![image](https://github.com/user-attachments/assets/2e798cbd-1e42-4846-a5c8-5d080b0b4e8d)
![image](https://github.com/user-attachments/assets/91dbe75b-0713-4497-8877-0fd1446e3d2f)
![image](https://github.com/user-attachments/assets/0ad77665-99dd-4c0e-b687-51ca97abe77d)

Hi
When I select USD, I can create a manual invoice in BTC Pay Server.

But indeed when I select ANG the BTC Pay Server shows a red error screen.

It doesn't match the ANG, Antillian Guilder currency.

1 USD is 1.78 ANG.

It has been fixed like that for years.

Shopify puts the currency as ANG because my location is Curacao.

ANG is used on Curacao and Sint Maarten.

Thanks for your help.

**Reply by pavlenex** [reply]
Which exchange/rate source you typically use for ANG? You may want to reach out to CoinGecko, since they have it but it doesn't seem to work.

**Comment by pavlenex** [comment]
@TChukwuleta ^

**Reply by pavlenex** [reply]
It seems ANG is not supported by CoinGecko. Perhaps you need to contact them?
<img width="688" alt="Screenshot 2024-12-17 at 11 23 11" src="https://github.com/user-attachments/assets/b4632041-7959-4102-8261-2f7903e8aab4" />

**Reply by RunyCalmera** [reply]
Hi would it be possible to download the BTC_USD rate from Kraken (that is my configuration now in BTC Pay server) and add a rule that 1 BTC_USD = 1.78 BTC_ANG?

Shopify has ANG as currency of Curacao. That is correct.

I don't know if I can set the shop currency to USD in shopify.

I will research that today.

It seems like BTC_ANG cannot be picked up.

For ANG if we know the BTC_USD we know the BTC_ANG, because we simply multiply it by 1.78.

So one way is to get BTC_USD from the exchange and set a rule manually to convert it to USD_ANG.

Thanks

Runy

**Reply by Kruwed** [reply]
>a rule that 1 BTC_USD = 1.78 BTC_ANG?

Currencies (even pegged ones) have a fluctuating market value, adding a static conversion rate is a bad idea that will eventually lead to one party overpaying/underpaying.

**Reply by RunyCalmera** [reply]
> > a rule that 1 BTC_USD = 1.78 BTC_ANG?
> 
> Currencies (even pegged ones) have a fluctuating market value, adding a static conversion rate is a bad idea that will eventually lead to one party overpaying/underpaying.

In the case of ANG, once the USD fluctuates, the ANG will fluctuate at the same rate, but then multiplied by 1.78. This currency has a fixed peg to the US Dollar.

The same as Aruban Guilder, which is used on Aruba.

Also the same as Eastern Caribbean Dollar. 1 USD = 2,7 Eastern Caribbean dollar.
These are 8 islands in the Caribbean.

So BTC Pay Server team can consider adding a feature to peg your currency to the USD maybe?

It will not fix all the problems.

But it will fix the problem of all the currencies that have a FIXED PEG to the dollar. These fixed pegs do not change. In the case of Curacao and Sint Maarten and Aruba these have been pegged during the last 40 years. For the Eastern Caribbean the peg has also been fixed during the past years.

How it could work is:
In BTC Pay Server where you retrieve the BTC_USD pair, you could add a feature: if your currency is not available on Kraken or Coingetco, consider searching for BTC_USD pair and then converting to your own currency. Only use this if your currency is a fixed peg to the USD.

Then people could fill in their currency like BTC_ANG. And add the 1.78.

This means that you do not need to change the base currency in your shopify shop to USD. You can still show the prices in your own currency ANG.

But BTC Pay Server will retrieve the current BTC_USD price from Kraken, and will then convert it to ANG by multiplying it with 1.78.


The path to ask Kraken or Coingetco to add BTC_ANG to their exchange is a lengthy one. Kraken will not add this because they do not see a lot of liquidity in that pair. And it will take very long to add that.

Runy

**Comment by RunyCalmera** [comment]
Hi all
I solved the problem.

I did this by setting the shopify store default currency to USD

https://community.shopify.com/c/payments-shipping-and/how-do-i-change-the-default-currency-on-my-site/m-p/2548625#:~:text=You%20can%20make%20changes%20to,navigating%20to%20the%20Help%20Center.


Basically if you have a currency that is not supported by Coingecko you need to set the basic currency to USD in Shopify.

BTC Pay server will then receive USD from Shopify and will retrieve the BTC_USD pair.

I was able to do this because you helped me researched this.

Thank you!

I will tell the people I know using ANG to from now on use USD in Shopify.

Thank you for your help.


Runy

![image](https://github.com/user-attachments/assets/612ded39-e8b2-4a3e-94f9-4704d1dad319)

![image](https://github.com/user-attachments/assets/0f9e648d-51c5-431c-9ef5-f04d57184760)
